### PR TITLE
DOC: fixed dynamic import mechanics of make.py

### LINKED ISSUE #20001: Unable to Build Documentation
```bash

python make.py html

Traceback (most recent call last):
  File "doc/make.py", line 23, in <module>
    import pandas
ModuleNotFoundError: No module named 'pandas'
```

I believe a slight bug was introduced by c8859b57b891701f250fb05f2cc60d2e6cae2d6b in including `import pandas` at the top of the script. Unless the user has pandas installed as a package I don't think this module would know where to find it without modifying the path or being more explicit on import

@jorisvandenbossche

### PR DESCRIPTION
**Pull request by WillAyd**
- [X] closes #20001 
- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/make.py [hunk 1]
```diff
@@ -332,7 +331,7 @@ def main():
                                  'compile, e.g. "indexing", "DataFrame.join"'))
     argparser.add_argument('--python-path',
                            type=str,
-                           default=os.path.join(DOC_PATH, '..'),
+                           default=os.path.dirname(DOC_PATH),
```
Comment (WillAyd) [inline doc/make.py]: This is purely stylistic

**Comment by jorisvandenbossche** [comment]
Looks good to me. I suppose this works for you?

cc @datapythonista

**Comment by datapythonista** [comment]
Yes, well spotted. Didn't realize the imported pandas wasn't being affected by the path.

**Comment by WillAyd** [comment]
Checked Travis failures and none look related to this change

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**